ivy_tests/test_ivy/test_frontends/test_tensorflow/test_general_functions.py

```diff
@@ -776,6 +776,37 @@ def test_tensorflow_shape_n(
     )
 
 
+@handle_frontend_test(
+    fn_tree="tensorflow.ensure_shape",
+    dtype_and_x=helpers.dtype_and_values(available_dtypes=helpers.get_dtypes("valid")),
+    shape=helpers.get_shape(
+        allow_none=True,
+        min_num_dims=0,
+        max_num_dims=10,
+        min_dim_size=0,
+        max_dim_size=10,
+    ),
+    output_dtype=st.sampled_from(["int32", "int64"]),
+)
+def test_tensorflow_ensure_shape(
+    *,
+    dtype_and_x,
+    shape,
+    fn_tree,
+    frontend,
+    test_flags,
+):
+    input_dtype, x = dtype_and_x
+    helpers.test_frontend_function(
+        input_dtypes=input_dtype,
+        frontend=frontend,
+        test_flags=test_flags,
+        fn_tree=fn_tree,
+        x=x[0],
+        shape=shape
+    )
+
```

Comment:
Not sure why you've added `output_dtype` and overall the test doesn't pass anyway. Since you're directly calling `set_shape` I would recommend you just copy the test for `set_shape` from `test_tensor` and import `_array_and_shape` into this file

Reply:
Fixed the test, i understand now how the tests are done, i was a bit confused and the test what template until i got testing working on my machine. thanks for the review